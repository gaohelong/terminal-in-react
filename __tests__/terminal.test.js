import React from 'react';
import renderer from 'react-test-renderer';
import PseudoFileSystem from 'terminal-in-react-pseudo-file-system-plugin'; // eslint-disable-line
import Terminal from '../components/Terminal';

const Term = (
  <Terminal
    backgroundColor="yellow"
    msg="Hello World. My name is Nitin Tulswani"
    commands={{
      'run': () => 'running something...'
    }}
    descriptions={{ run: 'Run something' }}
  />
);

const Term2 = (
  <Terminal
    plugins={[
      new PseudoFileSystem(),
    ]}
  />
);

describe('Terminal Component', () => {
  it ('should be a function', () => {
    expect(typeof Terminal).toBe('function');
  });

  it('should render a terminal component', () => {
    const tree = renderer.create(Term).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should work with plugins', () => {
    const tree = renderer.create(Term2).toJSON();

    expect(tree).toMatchSnapshot();
  });
});