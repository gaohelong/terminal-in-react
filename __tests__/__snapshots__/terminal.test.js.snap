// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terminal Component should render a terminal component 1`] = `
<div
  className="terminal-base"
  style={Object {}}
>
  <div
    className="terminal-container-wrapper"
    style={
      Object {
        "color": "green",
      }
    }
  >
    <div
      className="terminal-top-bar adjust-bar"
      style={
        Object {
          "backgroundColor": "black",
        }
      }
    >
      <svg
        height="20"
        width="100"
      >
        <circle
          cx="24"
          cy="14"
          fill="red"
          onClick={[Function]}
          r="5"
        />
        <circle
          cx="44"
          cy="14"
          fill="orange"
          onClick={[Function]}
          r="5"
        />
        <circle
          cx="64"
          cy="14"
          fill="green"
          onClick={[Function]}
          r="5"
        />
      </svg>
    </div>
    <div
      className="terminal-container terminal-container-main"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "yellow",
        }
      }
    >
      <div
        className="terminal-holder"
      >
        <div
          className="terminal-content"
        >
          <div
            className="terminal-input-area"
          >
            <pre
              className="terminal-output-line"
            >
              Hello World. My name is Nitin Tulswani
            </pre>
            <div
              className="terminal-input"
            >
              <span
                className="terminal-prompt"
                style={
                  Object {
                    "color": "green",
                  }
                }
              >
                &gt;
              </span>
              <input
                className="terminal-main-input"
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                style={
                  Object {
                    "backgroundColor": "yellow",
                    "color": "green",
                  }
                }
                tabIndex="-1"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Terminal Component should work with plugins 1`] = `
<div
  className="terminal-base"
  style={Object {}}
>
  <div
    className="terminal-container-wrapper"
    style={
      Object {
        "color": "green",
      }
    }
  >
    <div
      className="terminal-top-bar adjust-bar"
      style={
        Object {
          "backgroundColor": "black",
        }
      }
    >
      <svg
        height="20"
        width="100"
      >
        <circle
          cx="24"
          cy="14"
          fill="red"
          onClick={[Function]}
          r="5"
        />
        <circle
          cx="44"
          cy="14"
          fill="orange"
          onClick={[Function]}
          r="5"
        />
        <circle
          cx="64"
          cy="14"
          fill="green"
          onClick={[Function]}
          r="5"
        />
      </svg>
    </div>
    <div
      className="terminal-container terminal-container-main"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "black",
        }
      }
    >
      <div
        className="terminal-holder"
      >
        <div
          className="terminal-content"
        >
          <div
            className="terminal-input-area"
          >
            <div
              className="terminal-output-line"
            >
              Â 
            </div>
            <div
              className="terminal-input"
            >
              <span
                className="terminal-prompt"
                style={
                  Object {
                    "color": "green",
                  }
                }
              >
                /home/user/ &gt;
              </span>
              <input
                className="terminal-main-input"
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                style={
                  Object {
                    "backgroundColor": "black",
                    "color": "green",
                  }
                }
                tabIndex="-1"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
